cyctl/cmd/apply.go

```diff
@@ -0,0 +1,157 @@
+
+	"github.com/spf13/cobra"
+)
+
+var decUnstructured = yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+
+func doServerSideApply(ctx context.Context, cfg *rest.Config, obj *unstructured.Unstructured) error {
+	dc, err := discovery.NewDiscoveryClientForConfig(cfg)
+	if err != nil {
+		return err
+	}
+	mapper := restmapper.NewDeferredDiscoveryRESTMapper(memory.NewMemCacheClient(dc))
+
+	dyn, err := dynamic.NewForConfig(cfg)
+	if err != nil {
+		return err
+	}
+
+	mapping, err := mapper.RESTMapping(obj.GroupVersionKind().GroupKind(), obj.GroupVersionKind().Version)
+	if err != nil {
+		return err
+	}
+
+	var dr dynamic.ResourceInterface
+	if mapping.Scope.Name() == meta.RESTScopeNameNamespace {
+		dr = dyn.Resource(mapping.Resource).Namespace(obj.GetNamespace())
+	} else {
+		dr = dyn.Resource(mapping.Resource)
+	}
+
+	_, err = dr.Create(ctx, obj, metav1.CreateOptions{
+		FieldManager: "sample-controller",
+	})
+
+	return err
+}
+
+func applyYaml(yamlFile []byte, config *rest.Config) error {
+	multidocReader := utilyaml.NewYAMLReader(bufio.NewReader(bytes.NewReader(yamlFile)))
+
+	for {
+		buf, err := multidocReader.Read()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return err
+		}
+
+		obj := &unstructured.Unstructured{}
+		_, _, err = decUnstructured.Decode(buf, nil, obj)
+		if err != nil {
+			return err
+		}
+
+		err = doServerSideApply(context.TODO(), config, obj)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+var applyCmd = &cobra.Command{
+	Use:   "init",
+	Short: "initialize cyclops with all the resources (along with demo templates)",
+	Run: func(cmd *cobra.Command, args []string) {
+		version, err := cmd.Flags().GetString("version")
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		var deployUrl string
+		var demoUrl string
+
+		if version != "main" {
+			deployUrl = "https://raw.githubusercontent.com/cyclops-ui/cyclops/v" + version + "/install/cyclops-install.yaml"
+			demoUrl = "https://raw.githubusercontent.com/cyclops-ui/cyclops/v" + version + "/install/demo-templates.yaml"
+		} else {
+			deployUrl = "https://raw.githubusercontent.com/cyclops-ui/cyclops/" + version + "/install/cyclops-install.yaml"
+			demoUrl = "https://raw.githubusercontent.com/cyclops-ui/cyclops/" + version + "/install/demo-templates.yaml"
+		}
+
+		fmt.Println("using version:", version)
+
+		fmt.Println("fetching cyclops manifest")
+		deploy, err := http.Get(deployUrl)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		deployYamlFile, err := ioutil.ReadAll(deploy.Body)
+		if err != nil {
+			log.Printf("yamlFile.Get err   #%v ", err)
+		}
+
+		fmt.Println("initializing cyclops resources")
+
+		err = applyYaml(deployYamlFile, kubeconfig.Config)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		fmt.Println("initialization successful")
+
+		fmt.Println("fetching demo templates manifest")
+		demo, err := http.Get(demoUrl)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		defer deploy.Body.Close()
```

Comment:
Can you move this defer under this chunk of code? If an error happens while applying, it will not close the body since the defer is later in the code

```go
deploy, err := http.Get(deployUrl)
if err != nil {
    log.Fatal(err)
}
```